### PR TITLE
add-eslint-plugin-foreman

### PR DESCRIPTION
**Pull request by amirfefer**
I committed it manually because `npm run commit` doesn't work

* [ ] root
* [ ] @theforeman/builder
* [x] @theforeman/eslint-plugin-foreman
* [ ] @theforeman/env
* [ ] @theforeman/vendor
* [ ] @theforeman/vendor-dev
* [ ] @theforeman/vendor-core

## PR Type

* [ ] Bugfix
* [x] Feature
* [ ] Code style update (whitespace, formatting, missing semicolons, etc.)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Other… Please describe:

## Description

move linting config to foreman-js for resuing

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Does this PR fixes open issues?

* [ ] Yes
* [x] No

## Do you use this PR in another repository?

* [x] Yes
* [ ] No

### CODE REVIEW

File: packages/eslint-plugin-foreman/bin/tfm-lint.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+const cwd = process.cwd();
+const chalk = require('chalk');
+const rules = require('../lib/index.js');
+const { CLIEngine } = require('eslint');
+
+let cli = null;
+
+initCLI();
+const results = lintFiles(['/webpack', '/script']);
```
Comment (sharvit) [inline packages/eslint-plugin-foreman/bin/tfm-lint.js]: Shouldn't we receive the files list from the consumer? Maybe by a cli flag?

File: packages/eslint-plugin-foreman/bin/tfm-lint.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+/* eslint-disable no-console */
+
+const path = require('path');
+
+const cwd = process.cwd();
+const chalk = require('chalk');
+const rules = require('../lib/index.js');
+const { CLIEngine } = require('eslint');
+
+let cli = null;
```
Comment (sharvit) [inline packages/eslint-plugin-foreman/bin/tfm-lint.js]: Would it make sense to implement the `TfmLintCli` as a class?
```js
const linter = new TfmLintCli(files);
linter.lint();
```

Anyway, IMO some of the code here can move to the `lib` folder.

File: packages/eslint-plugin-foreman/bin/tfm-lint.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  });
+}
+
+function lintFiles(files) {
+  const filesList = files.map(file => `${cwd}/${file}`);
+  return cli.executeOnFiles(filesList);
+}
+
+function formatting(report) {
+  if (report.errorCount > 0) {
```
Comment (sharvit) [inline packages/eslint-plugin-foreman/bin/tfm-lint.js]: In case of an error, shouldn't we exit the script with an error code?

Comment (amirfefer) [reply packages/eslint-plugin-foreman/bin/tfm-lint.js]: these are linting errors, no runtime error

Comment (sharvit) [reply packages/eslint-plugin-foreman/bin/tfm-lint.js]: Once we have a linting error, the script should return a nonzero exit status so the CI will fail

**Comment by sharvit** [comment]
> I committed it manually because `npm run commit` doesn't work

It should work, please try to use it

**Pull request closed without merging**